docs/proposals/20210526-cluster-class-and-managed-topologies.md

```diff
@@ -194,588 +190,117 @@ If instead you are eager to see an example of ClusterClass and how the Cluster o
-  // schema. The schema is a subset of the schema used in
-  // Kubernetes CRDs.
-  OpenAPIV3Schema JSONSchemaProps `json:"openAPIV3Schema"`
-}
-```
-
-The schema implementation will be built on top of the [Open API schema embedded in Kubernetes CRDs](https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/types_jsonschema.go).
-To keep the implementation as easy and user-friendly as possible we will only implement the following feature set 
-for now (until further use cases emerge):
+This ClassClass CRD allows to define a collection of templates that describe the topology for one or more clusters.
+
+The detailed definition of this type can be found at [ClusterClass CRD reference](https://doc.crds.dev/github.com/kubernetes-sigs/cluster-api/cluster.x-k8s.io/ClusterClass/v1beta1);
+at high level the new CRD contains
+
+- The reference to the InfrastructureCluster template (e.g. AWSClusterTemplate) to be used when creating a Cluster using this ClusterClass
+- The reference to the ControlPlane template (e.g. KubeadmControlPlaneTemplate) to be used when creating a Cluster using this ClusterClass along with:
+  - The reference to infrastructureMachine template (e.g. AWSMachineTemplate) to be used when creating machines for the cluster's control plane.
+  - Additional attributes to be set when creating the control plane object, like metadata, nodeDrainTimeout, etc.
+  - The definition of a MachineHealthCheck to be created for monitoring control plane's machines.
+- The definition of how workers machines should look like in a Cluster using this ClusterClass, being composed of:
+  - A set of MachineDeploymentClasses, each one with: 
+    - The reference to the bootstrap template (e.g. KubeadmConfigTemplate) to be used when creating machine deployment machines.
+    - The reference to the infrastructureMachine template (e.g. AWSMachineTemplate) to be used when creating machine deployment machines.
+    - Additional attributes to be set when creating the machine deployment object, like metadata, nodeDrainTimeout, rolloutStrategy etc.
+    - The definition of a MachineHealthCheck to be created for monitoring machine deployment machines.
+- A list of patches, allowing to change above templates for each specific Cluster.
+- A list of variable definitions, defining a set of additional values the users can provide on each specific cluster;
+  those values can be used in patches. 
+
+The following paragraph provide some additional context on some of the above values; more info can
+be found in [writing a ClusterClass](https://cluster-api.sigs.k8s.io/tasks/experimental-features/cluster-class/write-clusterclass.html).
+
+**ClusterClass variable definitions**
+
+Each variable will be defined by providing an Open API schema using a type inspired from the [Open API schema](https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/types_jsonschema.go)
+which is embedded in Kubernetes CRDs definition.
```

Comment:
It might just be me, but this statement is not very clear. It is a little confusing.

How about something like: "Each variable can be defined by providing its own OpenAPI schema definition. The OpenAPI schema used is inspired from the schema used in Custom Resource Definitions in Kubernetes."